### PR TITLE
Xray pvc fix

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ x] Chart Version bumped
- [ x] CHANGELOG.md updated
- [ ] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/xray/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # JFrog Xray Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [3.0.21] - Mar 12, 2020
+* fix for xray pvc
```
Comment (eldada) [inline stable/xray/CHANGELOG.md]: Capital `F` in `Fix`.

File: stable/xray/templates/xray-statefulset.yaml [hunk 2]
```diff
@@ -356,9 +356,16 @@ spec:
...
           secretName: {{ template "xray.fullname" . }}-system-yaml
+      {{- if not .Values.server.persistence.enabled }}
       - name: data-volume
         emptyDir:
           sizeLimit: {{ .Values.server.persistence.size }}
+      {{- end }}
+      {{- if and .Values.server.persistence.enabled .Values.server.persistence.existingClaim }}
+      - name: data-volume
+        persistentVolumeClaim:
+          claimName: {{ .Values.server.persistence.existingClaim }}
```
Comment (eldada) [inline stable/xray/templates/xray-statefulset.yaml]: You are only supporting an existing PVC. We must work with `volumeClaimTemplates` as done in other charts.

File: stable/xray/CHANGELOG.md (hunk 1 above)
Comment (chukka) [reply stable/xray/CHANGELOG.md]: Done

File: stable/xray/templates/xray-statefulset.yaml (hunk 2 above)
Comment (chukka) [reply stable/xray/templates/xray-statefulset.yaml]: Added this change

File: stable/xray/templates/xray-statefulset.yaml [hunk 3]
```diff
@@ -356,17 +356,67 @@ spec:
...
+      {{ tpl .Values.server.customVolumes . | indent 6 }}
+      {{- end }}
       {{- if .Values.filebeat.enabled }}
       - name: filebeat-config
         configMap:
           name: {{ template "xray.fullname" . }}-filebeat-config
       {{- end }}
+  {{- if .Values.server.persistence.enabled }}
+  volumeClaimTemplates:
+  {{- if not .Values.server.persistence.existingClaim }}
```
Comment (eldada) [inline stable/xray/templates/xray-statefulset.yaml]: Don't restrict to `existingClaim`. You are forcing user to create one.

**Comment by evannorstrand** [comment]
Commenting here as well, this fix broke anyone trying to upgrade to this version and beyond, who were using the custom volumes code. Please see ticket #759

**Pull request merged**